Updated documentation here and there (#489)
newjersey/nj-ai-assistant: README-NJ.md

## How to Work in This Repo

### Spelunking LibreChat

AI Assistant is a fork of LibreChat, which at times is an intimidatingly large project.

Here are some tips for navigating such a large repository:

- The repo is compartmentalized into different modules. Focus on just one module at a time to avoid getting lost.

> - It is often easiest to find a React component either by searching for any visible text. The second easiest way is to
>   inspect the component then look for its CSS classes; that often narrows it down to just a handful of possible

Reviewer: dust/sand - This sounds a little un-grammatical, I would suggest:
```suggestion
- It is often easiest to find a React component either by searching for any visible text, or by
  inspecting the component and then looking for its CSS classes; that often narrows it down to just a handful of possible
```

1. **Contribute upstream, then later merge into our repo (preferred)**: this relies on the weekly upstream merge to
   capture our change. It's preferred because 1. you only need to make one PR and 2. if any changes are required for
   acceptance into upstream, they're handled before they get to our repo, minimizing merge conflicts.
2. **Simultaneously merge into our repo while contribution upstream**: if a change is needed ASAP in our repo, then we
   have no choice but to do it twice. Be wary of potential future merge conflicts!

### Adding New Models

We're using AWS Bedrock for all our model invocations.

When picking a model, make sure that you use an **in region inference** to keep requests in the US (e.g., instead of
`anthropic.claude-sonnet-4-6`, use `us.anthropic.claude-sonnet-4-6`).

Because we use these special **in region inference** models, we have to explicitly allow them to be used in the app.
That means you must update the comma-delimited env var `BEDROCK_AWS_MODELS` (in `nj.env.template` and locally for dev).

We use [model specs](https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/model_specs) to

> explicitly define the list of models users are allowed to access. These are defined in `nj-librechat.yaml`.

Reviewer 2: [dust] Maybe add that updating the above means we should update this yaml.

Any AIA conversation started with model "XYZ" is linked to model "XYZ". As such, you should never remove old models from

> the model specs (or else old conversations will no longer accept new prompts).

Reviewer 2: [dust] May be worth adding a warning icon or **Note** so this stands out.

2. **Simultaneously merge into our repo while contribution upstream**: if a change is needed ASAP in our repo, then we
   have no choice but to do it twice. Be wary of potential future merge conflicts!

### Adding New Models

We're using AWS Bedrock for all our model invocations.

When picking a model, make sure that you use an **in region inference** to keep requests in the US (e.g., instead of
`anthropic.claude-sonnet-4-6`, use `us.anthropic.claude-sonnet-4-6`).

Because we use these special **in region inference** models, we have to explicitly allow them to be used in the app.
That means you must update the comma-delimited env var `BEDROCK_AWS_MODELS` (in `nj.env.template` and locally for dev).

We use [model specs](https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/model_specs) to
explicitly define the list of models users are allowed to access. These are defined in `nj-librechat.yaml` and need to
be updated every time we add a new model

Warning: Any AIA conversation started with model "XYZ" is linked to model "XYZ". As such, **you should never remove old

> models from the model specs** (or else old conversations will no longer accept new prompts).

Reviewer: question - Woah, this is an intense limitation. I expect there will come a day when older models are deprecated and then removed in AWS. Just curious, do we have thoughts on what to do when that comes?

Author: I think that, at some point, we'll allow model switching which would (theoretically) let you upgrade a conversation to a new model. 

Though also, each conversation has a token limit, so there's a lifespan on any given convo.
